### PR TITLE
Notify followers when organizer creates new event

### LINKED ISSUE #443: Notify followers when organizer creates new event
Parent: #172
Depends on: #435

## Scope
Send notifications to followers when a user they follow creates a new event.

### 1. Notification Trigger
- Detect when a user creates a new event
- Fetch list of follower IDs from the organizer's profile
- Send notification to each follower

### 2. Notification Content
- Title: "New event from [Organizer Name]"
- Body: Event title and date
- Tapping notification opens event detail

### 3. Tests
- Unit tests for follower notification logic
- Integration test for notification delivery

## Acceptance Criteria
- [ ] Creating an event triggers notifications to organizer's followers
- [ ] Notification shows organizer name and event title
- [ ] Tapping notification opens the event
- [ ] Tests cover notification trigger and content

### PR DESCRIPTION
**Pull request by ziyad-elarroussi**
## Description
This PR implements follower notifications when a user they follow creates a new event. When an organizer creates an event, all their followers receive a push notification with the organizer's name and event title, allowing them to stay informed about events from people they follow.

**Related Issue:** Closes #443

---

## Changes

**Implementation:**
- Added `sendNewEventFromFollowedUserNotification()` method to `NotificationService` to handle follower notifications
- Integrated follower notification logic in `EventRepositoryFirestore.addEvent()` to trigger when events are created
- Fetch follower IDs from organizer's profile and send notifications to each follower
- Notification format: Title shows "New event from [Organizer Name]", body shows event title
- Tapping notification opens event detail screen via deep link

**Tests:**
- Added tests for notification content validation (title, message, type, metadata, actionUrl)
- Tested happy path where all followers receive notifications
- Tested edge cases: empty follower list, null profile, blank organizer name fallback
- All tests verify correct notification delivery and content formatting

---

## Testing
- [x] Unit tests pass
- [x] Instrumentation tests pass
- [x] Manual testing completed
- [x] Coverage maintained (≥80%)

**Test summary:** Added 4 comprehensive unit tests covering notification content, happy path with multiple followers, empty follower list handling, null profile handling, and organizer name fallback logic.

---

## Checklist
- [x] Sufficient documentation and minimal inline comments
- [x] All tests passing
- [x] No new warnings
- [x] If related issue exists: issue has all labels, fields, and milestone filled

### CODE REVIEW

File: app/src/main/java/com/swent/mapin/model/NotificationService.kt [hunk 1]
```diff
@@ -229,6 +229,41 @@ class NotificationService(
...
+    val notification =
+        Notification(
+            title = "New event from $organizerName",
+            message = eventTitle,
+            type = NotificationType.EVENT_INVITATION,
+            recipientId = recipientId,
+            senderId = organizerId,
+            readStatus = false,
+            metadata =
+                mapOf(
```
Comment (ItsaBox9368) [inline app/src/main/java/com/swent/mapin/model/NotificationService.kt]: Having metadata as a Map could potentially cause problems, since it could break silently and trigger exceptions from typos alone.

I suggest creating a data class for it:
```
data class EventNotificationMetadata(
    val eventId: String,
    val eventTitle: String,
    val organizerName: String
)
```
But this might be out of the scope of this PR, so take this as optional, I'm just not a big fan of having string literals which require precision down to the letter.

**Review by ItsaBox9368 — CHANGES_REQUESTED** [review]
Great job on this PR. It's a cool new feature that will be of great use.
The coverage is consistently good, the code quality as well. I have given some minor suggestions and nitpicks, some are optional. Keep up the good work.

File: app/src/main/java/com/swent/mapin/model/event/EventRepositoryFirestore.kt [hunk 2]
```diff
@@ -56,11 +57,15 @@ enum class UserEventSource(val fieldName: String) {
...
  * @param friendRequestRepository Repository for friend-related queries.
+ * @param notificationService Service for sending push notifications.
+ * @param userProfileRepository Repository for accessing user profile data.
  */
 class EventRepositoryFirestore(
     private val db: FirebaseFirestore,
     private val friendRequestRepository: FriendRequestRepository =
-        FriendRequestRepository(notificationService = NotificationService())
+        FriendRequestRepository(notificationService = NotificationService()),
+    private val notificationService: NotificationService = NotificationService(),
```
[the remark this answers is not in the document]
Comment (ziyad-elarroussi) [reply app/src/main/java/com/swent/mapin/model/event/EventRepositoryFirestore.kt]: Fixed- Removed default parameters and now injecting dependencies from EventRepositoryProvider.

File: app/src/test/java/com/swent/mapin/model/EventRepositoryFirestoreTest.kt [hunk 3]
```diff
@@ -1229,4 +1257,182 @@ class EventRepositoryFirestoreTest {
...
+    // Create repository with mocked dependencies
+    val mockNotificationService = mock<NotificationService>()
+    val mockUserProfileRepo = mock<UserProfileRepository>()
+
+    val repoWithNotifications =
+        EventRepositoryFirestore(
+            db,
+            friendRequestRepository = friendRepo,
+            notificationService = mockNotificationService,
+            userProfileRepository = mockUserProfileRepo)
```
[the remark this answers is not in the document]
Comment (ziyad-elarroussi) [reply app/src/test/java/com/swent/mapin/model/EventRepositoryFirestoreTest.kt]: Fixed - Extracted createRepoWithNotifications() helper to avoid duplication.

File: app/src/main/java/com/swent/mapin/model/NotificationService.kt (hunk 1 above)
Comment (ItsaBox9368) [reply app/src/main/java/com/swent/mapin/model/NotificationService.kt]: Sounds good, I agree

**Review by ItsaBox9368 — APPROVED** [review]
Sounds good ! The changes are good, approved.
